client/my-sites/plan-features-2023-grid/index.tsx

```diff
@@ -852,26 +865,76 @@ export class PlanFeatures2023Grid extends Component<
 			} );
 	}
 
+	renderStorageAddOnDropdown( properties: PlanProperties ) {
+		const { planName, storageOptions, storageFeatures } = properties;
+		const { selectedStorage } = this.state;
+
+		return (
+			<SelectDropdown
+				onToggle={ ( { open: isOpen }: { open: boolean } ) => {
+					if ( isOpen ) {
+						// record interaction event here
+					}
+				} }
+				selectedText={
+					selectedStorage[ planName ]
+						? getStorageStringFromFeature( selectedStorage[ planName ] )
+						: getStorageStringFromFeature( storageFeatures[ 0 ] )
+				}
+			>
+				{ storageOptions.map( ( storageOption ) => {
+					return (
+						<SelectDropdown.Item
+							key={ `${ planName } ${ storageOption }` }
+							selected={ storageOption === selectedStorage[ planName ] }
+							onClick={ () => {
+								const updatedSelectedStorage = {
+									...selectedStorage,
+									[ planName ]: storageOption,
+								};
+
+								this.setState( {
+									...this.state,
+									selectedStorage: updatedSelectedStorage,
+								} );
+							} }
+						>
+							{ getStorageStringFromFeature( storageOption ) }
+						</SelectDropdown.Item>
+					);
+				} ) }
+			</SelectDropdown>
+		);
+	}
+
 	renderPlanStorageOptions( planPropertiesObj: PlanProperties[], options?: PlanRowOptions ) {
-		const { translate } = this.props;
+		const { translate, intervalType } = this.props;
+
 		return planPropertiesObj
 			.filter( ( { isVisible } ) => isVisible )
 			.map( ( properties ) => {
 				if ( options?.isMobile && isWpcomEnterpriseGridPlan( properties.planName ) ) {
 					return null;
 				}
 
-				const { planName, storageOptions } = properties;
-				const storageJSX = storageOptions.map( ( storageFeature: string ) => {
-					if ( storageFeature.length <= 0 ) {
-						return;
-					}
-					return (
-						<div className="plan-features-2023-grid__storage-buttons" key={ planName }>
-							{ getStorageStringFromFeature( storageFeature ) }
-						</div>
-					);
-				} );
+				const { planName, storageOptions, storageFeatures } = properties;
+				const canUpgradeStorageForPlan =
+					storageOptions.length > 1 &&
+					intervalType === 'yearly' &&
+					isEnabled( 'plans/upgradeable-storage' );
```

Comment:
We should be grabbing and passing this value as a prop from `PlansFeaturesMain`. I don't think `isEnabled` is abstracted enough to work outside of Calypso environment i.e. when we migrate `plan-features-2023-grid` to a package, it can become unpredictable if putting together a storybook, for example. I think best to pass it from consumer end for now (plans-features-main).

Reply:
> I don't think isEnabled is abstracted enough to work outside of Calypso environment i.e. when we migrate plan-features-2023-grid to a package

Good callout -- I'll make the changes

Reply:
I added a simple boolean flag in [a929acd](https://github.com/Automattic/wp-calypso/pull/79041/commits/a929acd6ac727fff6e61e0aacfff047a4de6b998)